guides/cheatsheets/git.md

```diff
@@ -0,0 +1,305 @@
+repository synchronisation.
+
+You can download Git for Windows and Mac [here](https://git-scm.com/).
+
+If you prefer the Git Desktop client, navigate
+[here](https://desktop.github.com/).
+
+Alternative GUI programs are also available:
+
+- [SourceTree](https://www.sourcetreeapp.com/)
+- [Git Kraken](https://www.gitkraken.com/)
+- [VS Code Extension - GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens)
+
+---
+
+## **SETUP**
+
+---
+
+Configuring user information used across all local repositories.
+
+_Set a name that is identifiable you when review version history_.
+
+    git config --global user.name “[firstname lastname]”
+
+_Set an email address_.
+
+    git config --global user.email “[valid-email]”
+
+_Adding the `--local` option or not passing a config level option at all, will
+set the user.name for the current local repository._.
+
+    git config --local user.name "[First] [Last]"
+    git config --local user.email “[valid-email]”
+
+## **SETUP & INIT**
+
+Below are commands for initializing and cloning repositories.
+
+_Initialise an existing directory as a Git repository_.
+
+    git init
+
+_Retrieve an entire repository from a hosted location via URL_.
+
+    git clone [url]
+
+---
+
+## **STAGE & SNAPSHOT**
+
+---
+
+Working with snapshots and the Git staging area.
+
+_Show modified files in working directory, staged for your next commit_.
+
+    git status
+
+_Add a file as it looks now to your next commit (stage)_.
+
+    git add [file]
+
+_Add everything your next commit_.
+
+    git add .
+
+_Commit your staged content as a new commit snapshot_.
+
+    git commit -m “[descriptive message]”
+
+_Upload any changes to your local repository to your remote repository_.
+
+    git push
+
+_Upload any changes to your local repository to your remote repository,
+rewriting the git history (index)_.
+
+NOTE: It's normally discouraged except for certain use cases like squashing
+commits.
+
+    git push -f
+
+_If you find yourself in a situation where you are required to do a force push,
+you can push multiple branches and only force one like so: (+ denotes that the
+following branch will be force pushed (issue/1))_.
+
+    git push origin +issue/1 issue/2
+
+_Unstage a file while retaining the changes in working directory_.
+
+    git reset [file]
+
+_Diff of what is changed but not staged_.
+
+    git diff
+
+_Diff of what is staged but not yet commited_.
+
+    git diff --staged
+
+---
+
+## **BRANCH & MERGE**
+
+---
+
+Isolating work in branches, changing context, and integrating changes.
+
+_List your branches. a\*will appear next to the currently active branch\._
+
+    git branch
+
+_Create a new branch_.
+
+    git branch [branch-name]
+
+_Switch to another branch and check it out into your working directory_.
+
+    git checkout
+
+_Switch to the specified branch. Creates the branch in your local repository if
+it doesn't already exist_.
+
+    git checkout -b [branch_name]
```

Comment:
Worth noting that there's also `git branch` command for creating branches (but doesn't change branches as per the above; there's `git switch` for that too)?